src/core/segment_reader.rs

```diff
@@ -151,7 +152,10 @@ impl SegmentReader {
 
         let store_file = segment.open_read(SegmentComponent::Store)?;
 
-        fail_point!("SegmentReader::open#middle");
+        #[cfg(test)]
```

Comment:
Not sure if it is worth it, but you could define your own `fail_point!` macro in say `src/lib.rs` which forwards to `fail::fail_point!` for `#[cfg(test)]` and is a no-op otherwise. This would simplify these call sites and avoid the extra blocks.

Reply:
Yes that seems cleaner. Also I can't depend on `#[cfg(test)]` because test binaries don't get compiled with it (not sure why)